cmd/snap/cmd_snap_op.go

```diff
@@ -414,10 +414,15 @@ func (x *cmdInstall) installOne(name string, opts *client.SnapOptions) error {
 	} else {
 		changeID, err = cli.Install(name, opts)
 	}
-	if e, ok := err.(*client.Error); ok && e.Kind == client.ErrorKindSnapAlreadyInstalled {
-		fmt.Fprintf(Stderr, i18n.G("snap %q is already installed, see \"snap refresh --help\"\n"), name)
+	if e, ok := err.(*client.Error); ok {
+		msg, isError := clientErrorToCmdMessage(name, e)
+		if isError {
+			return errors.New(msg)
+		}
+		fmt.Fprintf(Stderr, msg)
```

Comment:
Yes, stderr is the right place to send warnings towards human eyes instead of the next tool in the pipeline. Strictly speaking, both cases are an error, actually. It's just that in one of them we have a text to present to the user instead of it being just a traditional error line.